src/tools/perf/cuda/ucp_cuda_kernel.cu

```diff
@@ -243,15 +245,18 @@ ucp_perf_cuda_send_async(const ucp_perf_cuda_params &params,
                          ucx_perf_counter_t idx, ucp_device_request_t *req,
                          ucp_device_flags_t flags = UCP_DEVICE_FLAG_NODELAY)
 {
+    const unsigned channel_id = threadIdx.x % params.num_channels;
```

Comment:
IMO the channel id should be some random value (or at least have a "mode" for channel where it's generated randomly)
channel_mode = enum { single, random, per-thread }
because in real scenario the "expert" index is "random"